agent/agent_worker.go

```diff
@@ -436,6 +436,8 @@ func (a *AgentWorker) runPingLoop(ctx context.Context, idleMonitor *IdleMonitor)
 		ranJob = true
 		if a.agentConfiguration.DisconnectAfterJob {
 			// Unless paused, this agent disconnects after the next ping.
+			// Do the ping immediately so we reduce the chances our agent is assigned a job
+			pingTicker.Reset(pingInterval)
```

Comment:
This same code is called immediately below, I thought about wrapping the `lastActionTime` in a `if` block instead, but it felt like that could cause issues if someone tries to add code that shouldn't happen here? 🤷

Reply:
I just thought of something that affects this (and the below) - we probably want to also bypass the jitter (select on line 344)

Reply:
yeah I realised this too, not sure of a good way to do that?

Reply:
Maybe:

1. above the loop `bypassJitter := make(chan struct{}, 1)`
2. in the jitter select, add a `case <-bypassJitter:`
3. when resetting the interval:

```go 
// Send on bypassJitter unless it's already got a value
select {
case bypassJitter <- struct{}{}:
default:
}
```